### ISSUE TITLE
shortcut key mappings get overwritten

### ISSUE THREAD

**Issue by kwkam** [visualboyadvance-m/visualboyadvance-m#120]
### Expected behavior
No unexpected config written to vbam.ini

### Actual behavior
Unexpected config written to vbam.ini

### Steps to reproduce the behavior
1. Compile without link support
2. Add these lines to vbam.ini
```
[Keyboard]
NOOP=ESC
```
3. Start visualboyadvance-m.exe and change an option to trigger `update_opts()` then exit
4. Open vbam.ini and see the Keyboard section
5. Repeat 3-4, the result would be like this:
```
[Keyboard]
NOOP=ESC
LanLink=ESC
<some_binary_data>=ESC
```

### Cause of the problem
1. At `wx/guiinit.cpp: 2590, 2602, 2614`, there are codes that modify `cmditem.cmd_id`
```
            if (cmdtab[i].cmd_id == XRCID("LanLink") || ...) {
                if (mi)
                    mi->GetMenu()->Remove(mi);

                cmdtab[i].cmd_id = XRCID("NOOP");
                cmdtab[i].mi = NULL;
                continue;
            }
```
2. At `wx/opts.cpp: 753`, there are codes that use `cmditem.cmd_id` for command detection, and after that, it does not check whether `cmd == ncmds` which may cause out of bounds read
```
        int cmd_id = i->GetCommand();
        int cmd;

        for (cmd = 0; cmd < ncmds; cmd++)
            if (cmdtab[cmd].cmd_id == cmd_id)
                break;

        ...

        if (nvs != cfg->Read(cmdtab[cmd].cmd))
            cfg->Write(cmdtab[cmd].cmd, nvs);
```

**Comment by Nicholas-Steel** [comment]
Can confirm something like this under Windows 10 Pro x64 v16299.214. Any changes I make to Shortcut keys is forgotten as soon as I close the program.

**Comment by denisfa** [comment]
Fixed in #439.
